### DISCUSSION TITLE
Tamagui in NX monorepo

### DISCUSSION THREAD

**Discussion by adamkrim** [tamagui/tamagui#2035] [General]
NX is popular for monorepo management and is an ideal place for tamagui !

Unfortunately, I'm not an expert in bundlers and compilers and I'm having trouble configuring a shared ui library containing tamagui with the best performance.

Like in this turborepo example [next-expo-solito](https://github.com/tamagui/tamagui/tree/master/starters/next-expo-solito), my goal is to setup an **ui shared library** containing and exporting a single tamagui configuration and custom components.

I believe that NX centralizes dependencies and builds at its root. I'm not sure what difference it makes to tamagui and his performance. I saw somme issues like #1565 but I don't know what to do in my repo.

I created a repository with a simple nx config [adamkrim/nx-tamagui](https://github.com/adamkrim/nx-tamagui) with 4 apps :
- nextjs
- react
- react-native
- expo

Can you help me find the most optimized configuration?

We could add the precise steps to the documentation for other interested NX users 🙏
I'm sorry that this request is related to my lack of knowledge. I'm still a junior and have never taken the time to learn about bundlers, compilers, ...

**Comment by kaceycleveland** [comment]
We have a similar stack for our org; I can take a stab at writing some documentation maybe on how to set things up.
I tried setting up some generic NX generators to create tamagui projects so maybe I will revisit that too.

**Comment by adamkrim** [comment]
@kaceycleveland It would be so cool if you can share with us your knowledge! 🙏
Just a simple buildable library example would be amazing

**Comment by Blendfoul** [comment]
For anyone still looking into this. 
This is a side project using nx with tamagui, that I'm doing for fun.

[Nx with Tamagui](https://github.com/Blendfoul/lfm-clone)

**Comment by guillempuche** [comment]
I created a repo https://github.com/guillempuche/nx-expo-next-tamagui that uses Nx, Expo React Native, NextJS, Tamagui & Solito.

The structure is: components, screens and apps (Expo and NextJS). NextJS still needs some work, but it works.

<img width="312" alt="imatge" src="https://github.com/tamagui/tamagui/assets/3519924/0ba46748-6b03-44a3-ba00-fcfe26808a68">

**Comment by natew** [comment]
ty guy, converting to discussion

**Comment by DevHussein** [comment]
Hi Team,
@natew 
I believe Nx is more widely known and used than Turborepo. Why don't we include an option in the Tamagui create template to choose between Turborepo and Nx?
